File: cmd/hatchet-cli/cli/templates/python/shared/src/workflows/first_workflow.py
from hatchet_sdk import Context

from hatchet_client import hatchet


# Declare the task to run
@hatchet.task(name="first-workflow")
def my_task(input: None, ctx: Context) -> dict[str, int]:
    print("executed task")

    return {"meaning_of_life": 42}

File: cmd/hatchet-cli/cli/templates/typescript/shared/src/workflows/first-workflow.ts
import { hatchet } from '../hatchet-client';

type Input = {
  message: string;
};

type Output = {
  "first-task": {
    message: string;
  };
  "second-task": {
    message: string;
  };
};

export const simple = hatchet.workflow<Input, Output>({
  name: 'first-workflow',
});

const step1 = simple.task({
  name: 'first-task',
  fn: (input) => {
    return {
      "message": "Hello, world!"
    };
  },
});

const step2 = simple.task({
  name: 'second-task',
  parents: [step1],
  fn: (input) => {
    return {
      "message": "Hello, moon!"
    };
  },
});
